src/plugins/binmap/binmap-format.js

```diff
@@ -0,0 +1,63 @@
+/// <reference types="@mapeditor/tiled-api" />
+
+/*
+ * binmap-format.js
+ *
+ * This extension adds the 'binary map format' type to the Export As menu.
+ * This is an alternative to the .so plugin, written in JavaScript.
+ * To install, copy to `~/.config/tiled/extensions`
+ *
+ * Simplest format possible. Consist of a 16 bytes long header and then data.
+ * See FORMAT.md for details.
+ */
+
+tiled.registerMapFormat("binmap", {
+	name: "Binary map files",
+	extension: "map",
+
+	write: (map, fileName) => {
+		let numLayers = 0, gid = 1;
+		/* fix missing firstgid properties on tilesets */
```

Comment:
This is not a "fix". "First global IDs" simply don't exist outside of the file format. It's fine to assign them as dynamic properties like this, but you'll want to reword that comment. :-)